script.js

```diff
@@ -1,12 +1,12 @@
-let todoInput, newTodo, ulList, completeBtn, deleteBtn, allBtn, activeBtn,completedBtn;
+let todoInput, newTodo, ulList, completeBtn, deleteBtn, allBtn, activeBtn,completedBtn, newTodos;
```

Comment:
- [ ] Reduce the scope of variables: Firstly, remove the global variable declarations and place them inside the functions where they're used. This can help reduce potential conflicts and bugs in larger applications.
- [ ] Combine related functions: prepareDOMElements() and prepareDOMEvents() have been combined into a single function prepareDOMElementsAndEvents(). This helps to group related logic together. In the combined function, we are querying the DOM for elements and immediately attaching event listeners to them.
- [ ] Use inline functions when appropriate: Instead of having a separate enterKeyCheck function, we have included its functionality directly inside the event listener for the 'keyup' event. If a function is only used once and its logic is relatively simple, it's often easier to understand when it's inline. This can be a matter of personal preference though.
- [ ] Pass arguments to functions: Instead of using global variables, pass arguments to functions where necessary. For example, addNewTodo now accepts todoInput and ulList as parameters. This makes functions more pure (i.e., they don't rely on outside state), which can make them easier to test and understand.
- [ ] Remove unused variables: If variables are declared but not used, like deleteBtn, completeBtn, allBtn, activeBtn, and completedBtn, they can be removed to simplify the code.
- [ ] Simplify conditionals: The if statement inside enterKeyCheck has been moved into the event listener for 'keyup' on todoInput, and combined with the check for whether todoInput.value is non-empty. This simplifies the code and makes it clearer what conditions are necessary to add a new todo.
- [ ] Add Comments: For a better understanding of the code functionality, add comments explaining what each function and event listener is doing.